python/ray/worker.py

```diff
@@ -264,6 +262,11 @@ def put_object(self, value, object_id=None, pin_object=True):
                 "do this, you can wrap the ray.ObjectID in a list and "
                 "call 'put' on it (or return it).")
 
+        if self.mode == LOCAL_MODE:
+            # TODO(ilr): Figure out how to make this work
+            assert object_id is None, ("Local Mode does not support "
+                                       "inserting with an objectID")
```

Comment:
I don't understand why inserting with an object id is hard?

Reply:
This was a problem when it was possible to have objects that were not Direct. Since this API (inserting with an object ID) is not public, I'm planning on not adding this functionality.